auth_server/server/server.go

```diff
@@ -38,6 +38,7 @@ import (
 
 var (
 	hostPortRegex = regexp.MustCompile(`\[?(.+?)\]?:\d+$`)
+	scopeRegex    = regexp.MustCompile(`([a-z0-9]+)($|\(([a-z0-9]+)\))`)
```

Comment:
`([a-z0-9]+)(\([a-z0-9]+\))?`

Reply:
Won't it produce the same result? Why do you prefer this other regex?

Reply:
yes, it better conveys the optional nature of type.